frontend/src/components/chat-input.tsx

```diff
@@ -77,14 +79,19 @@ export function ChatInput({
       );
       if (files.length > 0) {
         onImagePaste(files);
+        // Clear the text input when an image is dropped
+        onChange?.("");
       }
     }
   };
 
   const handleSubmitMessage = () => {
-    if (textareaRef.current?.value) {
-      onSubmit(textareaRef.current.value);
-      textareaRef.current.value = "";
+    if (value || (textareaRef.current?.value && !value)) {
+      onSubmit(value || textareaRef.current?.value || "");
+      if (!value && textareaRef.current) {
+        textareaRef.current.value = "";
+      }
+      onChange?.("");
```

Comment:
```suggestion
    if (textareaRef.current?.value) {
      onSubmit(textareaRef.current.value);
      textareaRef.current.value = "";
```

Undo this change for similar reasons as for `onChange`